src/clj/clj_diff/miller.clj

```diff
@@ -166,31 +166,35 @@
 (defn- next-edit
   "Find the next move through the edit graph which will decrease the
   edit distance by 1."
-  [a b graph delta p x k]
+  ([a b graph delta p x k]
+   (next-edit a b graph delta p x k {}))
+  ([a b graph delta p x k opts]
   {:post [(= (dec (edit-dist delta p k)) (:d %))]}
   (let [d (edit-dist delta p k)
-        head-x (backtrack-snake a b x (- x k))]
+         head-x (backtrack-snake a b x (- x k) opts)]
     (loop [head-x head-x]
       (let [move (first (filter #(and (not (= ::sentinel %)) ;; <<<===
                                       (= (:d %) (dec d)))
                                 (map #(% graph delta p head-x k)
                                      [look-left look-up])))]
         (if (and (< head-x x) (nil? move))
           (recur (inc head-x))
-          move)))))
+           move))))))
 
 (defn- edits
-  "Calculate the sequence of edits from the map of farthest reaching end
+  "Calculate the sequence of edits from the map of the farthest reaching end
   points."
-  [a b p delta graph]
-  (let [next-fn (partial next-edit a b graph delta)]
+  ([a b p delta graph]
+   (edits {} a b p delta graph))
+  ([opts a b p delta graph]
```

Comment:
This seems inconsistent with the addition of `opts` as the final arg in other functions, what's the reason for this?

Reply:
Yes, I know it looks like it but:
1. this function `edits` is private, not part of the public lib API so we can afford this kind of divergence. 
2. the technical reason of having the opts as the first argument is that the function is with `apply` in https://github.com/droitfintech/clj-diff/blob/dc3194799a5ace3e5a581ddd6412c06948fb3160/src/clj/clj_diff/miller.clj#L243 where it needs to have the varg arg on the last position.